engine/player/unique_gear_helper.hpp

```diff
@@ -336,7 +336,48 @@ struct proc_action_t : public T_ACTION
     if ( effect )
     {
       override_data( *effect );
+
+      // If ALL of the following are true:
+      //   1) the special_effect_t is an offensive spell action or an attack action
+      //   2) the proc action has 0 damage after parsing the spell data
+      //   3) the proc action has a trigger effect referencing the driver
+      //   4) the driver has a trigger effect referencing the proc action
+      // then the coefficient of the effect in 4) will be used for the damage of action 2)
+
+      bool is_dot = effect->duration() > 0_ms && effect->tick_time() > 0_ms;
+      auto type   = effect->action_type();
+
+      if ( ( type == SPECIAL_EFFECT_ACTION_SPELL || type == SPECIAL_EFFECT_ACTION_ATTACK ||
+             type == SPECIAL_EFFECT_ACTION_CUSTOM ) &&
+           ( ( is_dot && this->base_td == 0 ) || ( !is_dot && this->base_dd_min == 0 && this->base_dd_max == 0 ) ) )
+      {
+        for ( const auto& eff : this->data().effects() )  // go thru the driver's effects
+        {
+          auto trg = eff.trigger();
+
+          if ( trg->ok() )  // trigger'd damage spell found
```

Comment:
You could use init statements in the if conditional here and the one below at line 362 to make this a bit denser (but maybe instead spell out  the variable names some more? But that is just my personal preference not to shorten things that much :D)
E.g: 
```cpp
if ( const auto& trigger_spell = eff.trigger(); trigger_spell->ok() )
```